### PR TITLE
Add FadeAnimation

### LINKED ISSUE #35: [Proposal] Animation Behavior
# Animation Behavior

* [x] Proposed
* [x] Prototype
* [x] Implementation
  * [x] iOS Support
  * [x] Android Support
  * [x] macOS Support
  * [x] Windows Support
* [x] Unit Tests
* [x] Documentation: https://github.com/MicrosoftDocs/CommunityToolkit/pull/121
* [x] Sample: https://github.com/CommunityToolkit/Maui/blob/main/samples/CommunityToolkit.Maui.Sample/Pages/Behaviors/AnimationBehaviorPage.xaml

## Summary
[summary]: #summary

`AnimationBehavor` is a `Behavior` that can be attached to any `VisualElement`, allowing developers to trigger the following animations via `ICommand AnimateCommand`:
* [x] #519
* [ ] FlipHorizontalAnimation
* [ ] FlipVerticalAnimation
* [ ] RotateAnimation
* [ ] ScaleAnimation
* [ ] ShakeAnimation

## Motivation
[motivation]: #motivation

`AnimationBehavior` solves the following challenges:
- [x] Animations are not easily accomplished in XAML
- [x] Animations are not easily accomplished using `ICommand`

Following the implementation of `AnimationBehavior`, developers will easily be able to create bindable animations for any `VisualElement`

## Detailed Design
[design]: #detailed-design

### AnimationBehavior.shared.cs

`AnimationBehavior` is a `Behavior` can be attached to any `VisualElement`.

`AnimationBehavior` implements `EventToCommandBehavior`.

```cs
public class AnimationBehavior : EventToCommandBehavior
{
    public static readonly BindableProperty AnimationTypeProperty;
    public static readonly BindableProperty AnimateCommandProperty;

    public AnimationBase? AnimationType { get; set; }
    public ICommand AnimateCommand { get; }
}
```

### AnimationBase<TView>

`AnimationBase` is an `abstract` class implementing `BindableObject` that allows for creating animations.

```cs
public abstract class AnimationBase<TView> : BindableObject where TView : View
{
    public static readonly BindableProperty DurationProperty;
    public static readonly BindableProperty EasingTypeProperty

    public uint Duration { get; set; }
    public Easing Easing { get; set; }

    public abstract Task Animate(TView? view);
}
```

### LINKED ISSUE #519: FadeAnimation
Add in the `FadeAnimation` as a pre-built animation to be used with the `AnimationBehavior`.

The purpose of the `FadeAnimation` will be to fade to a specified opacity and then return to the original opacity over a specified duration (this applied to each animation).

* [x] Proposed
* [x] Prototype
* [x] Implementation
  * [x] iOS Support
  * [x] Android Support
  * [x] macOS Support
  * [x] Windows Support
* [x] Unit Tests
* [ ] Documentation: Not Started
* [x] Sample

### PR DESCRIPTION
**Pull request by bijington**
### Description of Change ###

 

 ### Linked Issues ###
 

 - Fixes #519

 ### PR Checklist ###
 
 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [x] Has tests (if omitted, state reason in description)
 - [x] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [x] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pull/121 

 ### Additional information ###

### CODE REVIEW

File: src/CommunityToolkit.Maui/Animations/FadeAnimation.shared.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+/// <summary>
+/// Animation that will fade the supplied view to the specified <see cref="Opacity"/>
+/// and then back to it's original <see cref="Opacity"/>.
+/// </summary>
+public class FadeAnimation : BaseAnimation
+{
+	/// <summary>
+	/// Initializes a new instance of <see cref="FadeAnimation"/>.
+	/// </summary>
+	public FadeAnimation() : base(300)
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui/Animations/FadeAnimation.shared.cs:12]: what is 300 here? is it duration?

**Review by VladislavAntonyuk — DISMISSED** [review]

File: src/CommunityToolkit.Maui/Animations/FadeAnimation.shared.cs (hunk 1 above)
Comment (brminnick) [reply src/CommunityToolkit.Maui/Animations/FadeAnimation.shared.cs:12]: It's the default length of the animation: https://github.com/CommunityToolkit/Maui/blob/5ade8c29f906e08235fe5996b6d8d173ff74ac36/src/CommunityToolkit.Maui/Animations/BaseAnimation.shared.cs#L61

**Review by rarewater1 — DISMISSED** [review]

**Review by brminnick — APPROVED** [review]
Approving again following `dotnet format`

**Pull request merged**